### PR TITLE
9.2.4 support for plutus-ledger-api

### PR DESCRIPTION
**Pull request by michaelpj**

### CODE REVIEW

File: __std__/cells/plutus/library/agda-packages.nix [hunk 1]
```diff
@@ -36,7 +36,7 @@ let
     {
       haskellPackages = pkgs.haskellPackages //
       {
-        ghcWithPackages = cell.library.plutus-project.ghcWithPackages;
+        ghcWithPackages = (cell.library.plutus-project { }).ghcWithPackages;
```
Comment (zeme-iohk) [inline __std__/cells/plutus/library/agda-packages.nix:39]: I'd love to be able to get rid of these `(cell.library.plutus-project { })`
To do that, we can
1. Rename `__std__/cells/plutus/library/plutus-project.nix` -> `__std__/cells/plutus/library/make-plutus-project.nix` 
2. Create `__std__/cells/plutus/library/plutus-project-ghc8107.nix`
    ```
    { inputs, cell }:
    cell.library.make-plutus-project { compiler-nix-name = "ghc8107" } 
    ```
3. Create `__std__/cells/plutus/library/plutus-project-ghc942.nix`
    ```
    { inputs, cell }:
    cell.library.make-plutus-project { compiler-nix-name = "ghc942" } 
    ```

Comment (zeme-iohk) [reply __std__/cells/plutus/library/agda-packages.nix:39]: It would also make it very explicit where we are using which

Comment (zeme-iohk) [reply __std__/cells/plutus/library/agda-packages.nix:39]: Then we could also get rid of the file `__std__/cells/plutus/library/ghc-compiler-nix-name.nix`

File: __std__/cells/plutus/library/plutus-project.nix [hunk 2]
```diff
@@ -190,7 +190,7 @@ let
   });
 
 in
-
-project.appendOverlays [
+{ compiler-nix-name ? library.ghc-compiler-nix-name }:
```
Comment (zeme-iohk) [inline __std__/cells/plutus/library/plutus-project.nix:200]: if you move this to line 2, then `project` on line 20 does not need the `{ compiler-nix-name }:`

File: __std__/cells/plutus/library/agda-packages.nix (hunk 1 above)
Comment (michaelpj) [reply __std__/cells/plutus/library/agda-packages.nix:39]: Hmmmm, I guess that's fine yes.

File: __std__/cells/plutus/library/make-plutus-project.nix [hunk 3]
```diff
@@ -0,0 +1,201 @@
+{ inputs, cell }:
+{ compiler-nix-name ? inputs.cells.toolchain.library.ghc-compiler-nix-name }:
```
Comment (zeme-iohk) [inline __std__/cells/plutus/library/make-plutus-project.nix:2]: the default `? inputs.cells.toolchain.library.ghc-compiler-nix-name` is never triggered now, do we want to remove that file?

Comment (michaelpj) [reply __std__/cells/plutus/library/make-plutus-project.nix:2]: Yeah, I guess it's not that useful indeed.

Comment (michaelpj) [reply __std__/cells/plutus/library/make-plutus-project.nix:2]: It's still used in a few places. I'll leave it for now.

**Comment by michaelpj** [comment]
Okay, I can build `plutus-ledger-api` with both 8.10.7 and 9.2.4 locally, so hopefully the CI will confirm and we're good to go.

File: __std__/cells/automation/ciJobs.nix [hunk 4]
```diff
@@ -41,9 +56,14 @@ let
...
     if system == "x86_64-linux" then
-      { mingwW64 = windows-plutus-jobs; } // native-plutus-jobs // other-jobs
+      { mingwW64 = windows-plutus-jobs; } //
+      native-plutus-jobs //
+      { ghc924 = native-plutus-924-jobs; } //
+      other-jobs
     else if system == "x86_64-darwin" then
-      native-plutus-jobs // other-jobs
+      native-plutus-jobs //
+      native-plutus-924-jobs //
```
Comment (zeme-iohk) [inline __std__/cells/automation/ciJobs.nix:65]: we want `{ ghc924 = native-plutus-924-jobs; }` here

Comment (michaelpj) [reply __std__/cells/automation/ciJobs.nix:65]: argument for deduplicating this and writing
```
jobs = x // y // lib.optionalAttrs (system == "x86_64-linux") { mingwW64 = ... } // ...
```

**Review by zeme-iohk — APPROVED** [review]
Sorry I just merged my other nix PR so you'll have to rebase

**Comment by michaelpj** [comment]
Manual benchmark run results (8.10 is old, 9.2.4 is new):
```
| Script                                   |  Old       |  New       |   Change   |                                                                                                                                                                                                                                        
| :------                                  |  :------:  |  :------:  |  :------:  |                                                                                                                                                                                                                                        
| auction_1-1                              |  184.3 μs  |  194.1 μs  |    +5.3%   |                                                                                                                                                                                                                                        
| auction_1-2                              |  778.4 μs  |  705.2 μs  |    -9.4%   |                                                                                                                                                                                                                                        
| auction_1-3                              |  767.4 μs  |  699.2 μs  |    -8.9%   |                                                                                                                                                                                                                                        
| auction_1-4                              |  233.4 μs  |  248.5 μs  |    +6.5%   |                                                                                                                                                                                                                                        
| auction_2-1                              |  185.3 μs  |  195.3 μs  |    +5.4%   |                                                                                                                                                                                                                                        
| auction_2-2                              |  781.4 μs  |  707.1 μs  |    -9.5%   |                                                                                                                                                                                                                                        
| auction_2-3                              |  994.6 μs  |  913.0 μs  |    -8.2%   |                                                                                                                                                                                                                                        
| auction_2-4                              |  771.4 μs  |  700.1 μs  |    -9.2%   |                                                                                                                                                                                                                                        
| auction_2-5                              |  233.0 μs  |  249.5 μs  |    +7.1%   |                                                                                                                                                                                                                                        
| crowdfunding-success-1                   |  215.9 μs  |  227.9 μs  |    +5.6%   |                                                                                                                                                                                                                                        
| crowdfunding-success-2                   |  215.6 μs  |  227.5 μs  |    +5.5%   |                                                                                                                                                                                                                                        
| crowdfunding-success-3                   |  215.6 μs  |  227.7 μs  |    +5.6%   |                                                                                                                                                                                                                                        
| currency-1                               |  281.5 μs  |  278.4 μs  |    -1.1%   |                                                                                                                                                                                                                                        
| escrow-redeem_1-1                        |  400.1 μs  |  382.0 μs  |    -4.5%   |                                                                                                                                                                                                                                        
| escrow-redeem_1-2                        |  400.4 μs  |  383.2 μs  |    -4.3%   |
| escrow-redeem_2-1                        |  470.0 μs  |  447.3 μs  |    -4.8%   |
| escrow-redeem_2-2                        |  470.2 μs  |  445.7 μs  |    -5.2%   |
| escrow-redeem_2-3                        |  471.0 μs  |  445.5 μs  |    -5.4%   |
| escrow-refund-1                          |  164.0 μs  |  168.2 μs  |    +2.6%   |
| future-increase-margin-1                 |  282.0 μs  |  278.1 μs  |    -1.4%   |
| future-increase-margin-2                 |  644.9 μs  |  587.2 μs  |    -8.9%   |
| future-increase-margin-3                 |  643.8 μs  |  587.4 μs  |    -8.8%   |
| future-increase-margin-4                 |  602.0 μs  |  536.6 μs  |   -10.9%   |
| future-increase-margin-5                 |  973.6 μs  |  884.2 μs  |    -9.2%   |
| future-pay-out-1                         |  280.9 μs  |  277.6 μs  |    -1.2%   |
| future-pay-out-2                         |  643.9 μs  |  590.2 μs  |    -8.3%   |
| future-pay-out-3                         |  643.7 μs  |  585.7 μs  |    -9.0%   |
| future-pay-out-4                         |  971.3 μs  |  886.1 μs  |    -8.8%   |
| future-settle-early-1                    |  280.6 μs  |  279.5 μs  |    -0.4%   |
| future-settle-early-2                    |  642.2 μs  |  592.1 μs  |    -7.8%   |
| future-settle-early-3                    |  640.8 μs  |  590.3 μs  |    -7.9%   |
| future-settle-early-4                    |  742.0 μs  |  666.0 μs  |   -10.2%   |
| game-sm-success_1-1                      |  452.1 μs  |  423.3 μs  |    -6.4%   |
| game-sm-success_1-2                      |  199.8 μs  |  215.7 μs  |    +8.0%   |
| game-sm-success_1-3                      |  765.3 μs  |  694.4 μs  |    -9.3%   |
| game-sm-success_1-4                      |  232.3 μs  |  248.2 μs  |    +6.8%   |
| game-sm-success_2-1                      |  453.8 μs  |  422.6 μs  |    -6.9%   |
| game-sm-success_2-2                      |  199.7 μs  |  215.0 μs  |    +7.7%   |
| game-sm-success_2-3                      |  764.4 μs  |  697.6 μs  |    -8.7%   |
| game-sm-success_2-4                      |  231.4 μs  |  248.6 μs  |    +7.4%   |
| game-sm-success_2-5                      |  763.0 μs  |  696.8 μs  |    -8.7%   |
| game-sm-success_2-6                      |  231.6 μs  |  250.0 μs  |    +7.9%   |
```
Things have definitely Happened here. Unclear if there's noise going around or what. I think given it's not consistently worse or anything I wouldn't be devastated if they released this, so I think we can merge for now and then take a look at the core later.

**Comment by michaelpj** [comment]
Well, I'll take a look now for curiosity at least.

File: CONTRIBUTING.adoc [hunk 5]
```diff
@@ -172,6 +172,18 @@ and https://nixos.org/manual/nix/unstable/command-ref/new-cli/nix3-flake-lock.ht
 Specifically, you will probably want to say `nix flake lock --update-input <input-name>`.
 Do *not* use `nix flake update`, as that will update all the inputs, which we typically don't want to do.
 
+== GHC versions
+
+=== 9.2 migration
+
+Our primary major version of GHC is 8.10. 
+However, the Cardano projects are in the process of moving to 9.4.
```
Comment (zeme-iohk) [inline CONTRIBUTING.adoc:180]: Is it 9.2 or 9.4?

Comment (michaelpj) [reply CONTRIBUTING.adoc:180]: :facepalm:

File: __std__/cells/automation/ciJobs.nix (hunk 4 above)
Comment (zeme-iohk) [reply __std__/cells/automation/ciJobs.nix]: 👍

**Pull request merged**